src/items/FavoriteButton.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { FC, MouseEventHandler } from 'react';
+import { IconButton, Tooltip } from '@material-ui/core';
+import { useTranslation } from 'react-i18next';
+import StarIcon from '@material-ui/icons/Star';
+import StarBorderIcon from '@material-ui/icons/StarBorder';
+import { FAVORITE_ITEM_BUTTON_CLASS } from '../constants';
+
+export interface FavoriteButtonProps {
+  isFavorite: boolean;
+  handleUnfavorite: MouseEventHandler;
+  handleFavorite: MouseEventHandler;
+  size: 'default' | 'small' | 'large' | 'inherit' | 'medium' | undefined;
+  color: 'primary' | 'secondary' | 'default' | undefined;
```

Comment:
Nice, is there any possibility to use a type from Material directly? It would be more dynamic.